ragna/source_storages/_qdrant.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
+
+import os
+import uuid
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Optional, cast
+
+import ragna
+from ragna.core import (
+    Document,
+    MetadataFilter,
+    MetadataOperator,
+    PackageRequirement,
+    Requirement,
+    Source,
+)
+
+from ._utils import raise_no_corpuses_available, raise_non_existing_corpus
+from ._vector_database import VectorDatabaseSourceStorage
+
+if TYPE_CHECKING:
+    from qdrant_client import models
+
+
+class QdrantDB(VectorDatabaseSourceStorage):
+    """[Qdrant vector database](https://qdrant.tech/)
+
+    !!! info "Required packages"
```

Comment:
Could you add something here that indicates that the user might need to set the `QDRANT_URL` and `QDRANT_API_KEY` env vars?

Reply:
Resolved with https://github.com/Quansight/ragna/pull/524/commits/c9d19e7b967f2151c0b66d93682aef51b4161239.